admin/client/App/screens/Item/actions.js

```diff
@@ -23,13 +23,26 @@ export function selectItem (itemId) {
  */
 export function loadItemData () {
 	return (dispatch, getState) => {
+		// Crete an empty object to reference a point in memory.
+		// Dispatch this reference to our redux store to hold on to as a 'loadingRef'.
+		const thisLoadRef = {};
 		dispatch({
 			type: LOAD_DATA,
+			loadingRef: thisLoadRef,
 		});
 		const state = getState();
 		const list = state.lists.currentList;
 		// Load a specific item with the utils/List.js helper
 		list.loadItem(state.item.id, { drilldown: true }, (err, itemData) => {
+			// Once this async request has fired this callback, check that
+			// the point in memory referenced by thisLoadRef is the same point in memory
+			// referenced by loadingRef in the redux store.
+
+			// If it is, then this is the latest request, and it is safe to resolve it normally.
+			// If it is not a reference the same point in memory however,
+			// this means that this request is NOT the latest fired request,
+			// and so we'll bail out of it early.
+			if (getState().item.loadingRef !== thisLoadRef) return;
```

Comment:
This isn't about optimisation - if you were to fire off two async load requests for the same item, only the last should actually update the store. If you want a simpler unique check, we can use an increment prefixed by the item ID but that seemed more complex to me when I first added this.

I guess fundamentally I'm not aware of the issues this could cause, and I'm surprised that it's a problem given that (as I understand it) this method is the basis for the whole immutable lib. It makes reference checking a valid way to assert deep equality. Surely if that doesn't work with Redux, it would be a massive problem?

Why would the devtools recreate objects in state? Unless we're talking about exporting / importing state, I guess?